src/agent/include/unix_daemon.hpp

```diff
@@ -1,3 +1,22 @@
 #pragma once
 
-#include <process_options.hpp>
+#include <string_view>
+#include <unistd.h>
+#include <vector>
+
+class UnixDaemon
```

Comment:
Just to iterate over this class... do you think it represents a `UnixDaemon`? From what it is doing seems to be just a convenience RAII wrapper over the creation of a PID file, plus the `GetStatus` functionality.

An idea that comes to mind is to have a PIDFileHandler that does exactly what the UnixDaemon is doing in its constructor/destructor. This can be a nested class inside the UnixDaemon class, or in the same namespace. Then we can have a static function called CreatePIDFile in the UnixDaemon class or namespace, much like what you have for the status, that returns this handler, guaranteeing the lifetime of the file as you intended.

But now when you use it instead of creating a unix daemon instance that is apparently unused, you explicitly create the handler:

```cpp
const auto pidFileHandler = UnixDaemon::CreatePidFile();
Agent agent;
agent.Run();
```

Just a thought, happy to discuss it further, or to drop it if you don't see it fit.

Reply:
One more thing, what happens with the PID files if the application crashes? Should the stale files be cleaned before creating new ones?

Reply:
Done. Changed class UnixDaemon for PIDFileHandler and added two functions in the namespace to create the pid file and to get the status.
Since we now have only one daemon, we are simply overwriting any existing file in writePIDFile().